### DISCUSSION TITLE
SQLite attach support

### DISCUSSION THREAD

**Discussion by Nvos** [ent/ent#1419] [General]
I would like to know if it would be possible to use attached database (https://www.sqlite.org/lang_attach.html). It seems that there's multi schema support but but after some digging found out that it is disabled for SQLite.

**Comment by rotemtam** [comment]
Hi @Nvos !
Thanks for reaching out. Can you explain more about the use case, motivation and what would you expect from such an API? 

Thanks

**Reply by Nvos** [reply]
Use case is possibility of attaching in-memory database to file sqlite database, this allows for handling some tables as in-memory while having single source for data.

Motivation for this request is that currently in one of desktop apps we are storing some data in memory storages this data is frequently changing and is temporary so we do not want to store it persistently but quite often we have to combine/filter data from persistent and memory sources.

I'm unsure if any new API is needed to handle this, quite likely it could be handled internally when feature https://entgo.io/docs/feature-flags#schema-config is enabled (currently prefixing with schema is disabled for sqlite). I don't think that attaching database has to be part of ent it can be done manually when opening connection. Only thing which would be needed is possibility of running migration on specific schema(which in case of sqlite would be attached db)
